### PR TITLE
Custom Feedback Paths: Update Delete Cascades

### PR DESCRIPTION
**Pull request by JosephineKwa**
1. Cascaded student email, instructor email and team name changes to feedback paths
2. Cascaded student, instructor deletion to feedback paths
3. Cascaded feedback path deletion to responses

### CODE REVIEW

File: src/test/java/teammates/test/cases/logic/StudentsLogicTest.java [hunk 1]
```diff
@@ -1105,8 +1105,8 @@ private static StudentEnrollDetails enrollStudent(StudentAttributes student) thr
                                                    StudentsLogic.inst(), new Object[] { student, false });
     }
 
-    @AfterClass
-    public void classTearDown() {
+    @AfterClass()
```
Comment (chowyb) [inline src/test/java/teammates/test/cases/logic/StudentsLogicTest.java]: Not sure why this was changed to use brackets

Comment (JosephineKwa) [inline src/test/java/teammates/test/cases/logic/StudentsLogicTest.java]: Oops, my mistake

File: src/main/java/teammates/logic/core/FeedbackResponsesLogic.java [hunk 2]
```diff
@@ -687,6 +687,65 @@ public void deleteFeedbackResponseAndCascade(FeedbackResponseAttributes response
     }
 
     public void deleteFeedbackResponsesForQuestionAndCascade(
+            FeedbackQuestionAttributes oldQuestion, FeedbackQuestionAttributes newQuestion,
+            boolean hasResponseRateUpdate) {
```
Comment (JosephineKwa) [inline src/main/java/teammates/logic/core/FeedbackResponsesLogic.java]: Would the following javadoc be alright?
```
/**
 * Deletes feedback responses for an updated question.
 * newQuestion is compared with oldQuestion to determine the responses to be deleted
 * based on the changes in feedback paths.
 */
```

Comment (chowyb) [inline src/main/java/teammates/logic/core/FeedbackResponsesLogic.java]: Seems fine :+1:

**Comment by damithc** [comment]
@whipermr5 I didn't review previous PRs on this long-lived branch, so I wont review this one either. I'll take a look when the branch is ready to merge.

**Comment by whipermr5** [comment]
@damithc Did you mean to mention @JosephineKwa?

**Comment by damithc** [comment]
> @damithc Did you mean to mention @JosephineKwa?

Sorry, I meant @chowyb (he's the one who requested the review :-)

**Comment by JosephineKwa** [comment]
@damithc is it alright if I merge this PR?

**Comment by damithc** [comment]
> @damithc is it alright if I merge this PR?

Yes @JosephineKwa , you can merge to the feature branch.

**Pull request merged**